tests/test_peer_heap.py

```diff
@@ -107,6 +107,18 @@ def test_update(peer_heap, peers):
     assert peer_heap.peek_peer() == p
 
 
+def test_remove(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = len(peers)
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+
```

Comment:
Add tests for at least the following cases:

- double removal: remove_peer on the same peer twice
- removing a peer from an empty peer heap
- removing a peer that doesn't exist in the heap